src/draws/drawAxisComponents.js

```diff
@@ -70,9 +70,13 @@ export default (function drawAxisComponents(
         ) {
           gatherTextTriangles(store, viz_area, inst_axis);
         }
-        regl(text_triangle_args)(
-          store.getState().visualization.text_triangles.draw[inst_axis]
-        );
+        // we have to check for existence of text vectors to draw now, as they
+        // might be created async and we might not have all of them
+        if (state.visualization.text_triangles.draw[inst_axis] !== false) {
```

Comment:
We should DRY this with the one that we're sending to regl, and use `store.getState()` just so we're sure we have the latest data.